jobserver/views/projects.py

```diff
@@ -153,24 +156,27 @@ def get_status(self, project):
             "variant": variants_lut[project.status],
         }
 
-    def iter_repos(self, repos, tracer):
-        @tracer.start_as_current_span("get_repo")
-        def get_repo(repo):
-            try:
-                is_private = self.get_github_api().get_repo_is_private(
-                    repo.owner, repo.name
-                )
-            except requests.HTTPError:
-                is_private = None
-
-            return {
-                "name": repo.name,
-                "url": repo.url,
-                "is_private": is_private,
-            }
+    def iter_repos(self, repos, ctx):
+        def get_repo(repo, ctx):
+            otel_context.attach(ctx)
+            with self.tracer.start_as_current_span("get_repo"):
+                try:
+                    is_private = self.get_github_api().get_repo_is_private(
+                        repo.owner, repo.name
+                    )
+                except requests.HTTPError:
+                    is_private = None
+
+                return {
+                    "name": repo.name,
+                    "url": repo.url,
+                    "is_private": is_private,
+                }
```

Comment:
we can attach some attributes to the span, like:
```suggestion
                try:
                    is_private = self.get_github_api().get_repo_is_private(
                        repo.owner, repo.name
                    )
                except requests.HTTPError:
                    is_private = None

                span = trace.get_current_span()
                span.set_attribute("repo_owner", repo.owner)
                span.set_attribute("repo_name", repo.name)
                span.set_attribute("repo_is_private", repo.is_private)

                return {
                    "name": repo.name,
                    "url": repo.url,
                    "is_private": is_private,
                }
```